modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/consensus/ConsensusStateUpdater.scala

```diff
@@ -345,6 +373,20 @@ object ConsensusStateUpdater {
       }
     }
 
+    private def checkFacilitatorsHashes[A](facilitatorsHash: A => Hash, ownFacilitatorsHash: Option[Hash])(
+      declarations: List[A]
+    ): F[Option[List[A]]] =
+      pickMajority(declarations.map(facilitatorsHash)).filter { majorityFacilitatorsHash =>
+        ownFacilitatorsHash.map(_ === majorityFacilitatorsHash).getOrElse(true)
+      }
+        .map(_ => declarations)
+        .pure[F]
+        .flatTap(declarations =>
+          logger
+            .warn(s"Different facilitators hashes. This node is in fork")
+            .whenA(declarations.isEmpty)
```

Comment:
99% of people reading this code will think that `declarations.isEmpty` checks if the list of declarations is empty